lib/internal/modules/cjs/loader.js

```diff
@@ -966,10 +973,15 @@ function getExportsForCircularRequire(module) {
  * 3. Otherwise, create a new module for the file and save it to the cache.
  *    Then have it load the file contents before returning its exports object.
  * @param {string} request Specifier of module to load via `require`
- * @param {string} parent Absolute path of the module importing the child
+ * @param {Module} parent Absolute path of the module importing the child
  * @param {boolean} isMain Whether the module is the main entry point
  */
 Module._load = function(request, parent, isMain) {
+  const logLabel = `[${parent?.id || ''}] [${request}]`;
+  const traceLabel = `require('${request}')`;
+
+  startTimer(logLabel, traceLabel);
```

Comment:
Shouldn't it check if trace events are enabled first? It seems to affect (in terms of performance) all users in general, not just NODE_DEBUG

Reply:
I do the check at https://github.com/nodejs/node/pull/52213/files#diff-2b09f603b3132191141319f8bfa6d5af325d40d0827a9fc10f1757af9fe21cafR218-R220

Since there are two conditions to enable (log and trace), I need to check it later.

Reply:
The typical pattern of `debuglog` and friend is, when NODE_DEBUG is enabled, on first invocation the callback of `debugWithTimer` would replace `startTimer` with a function that does something. Otherwise it's replaced with noops which V8 can inline away.

Reply:
Since we are integrating with `trace_events`, there's no way to avoid perf penalty here and without having a listener to the `trace_events` (I don't think that exist), we can't replace with `noop` functions.

Reply:
Ok, as soon I have time I will rewrite this to change to noop in the initialization, then I will create a PR to fix this issue based on https://github.com/nodejs/node/pull/48142

Reply:
Yes I think a use case like this would justify something like https://github.com/nodejs/node/pull/48142 - doing the trace enabling check is going to lead to a bigger perf hit on module loading than it does on HTTP (it didn't actually look like that much code to me, BTW, it's mostly just boilerplate binding code, but maybe I am biased because I am too used to boilerplate binding code).